tinygrad/nn/__init__.py

```diff
@@ -24,7 +25,7 @@ def __call__(self, x:Tensor):
       # NOTE: wow, this is done all throughout training in most PyTorch models
       if self.track_running_stats:
         self.running_mean.assign((1 - self.momentum) * self.running_mean + self.momentum * batch_mean.detach())
-        self.running_var.assign((1 - self.momentum) * self.running_var + self.momentum * batch_var.detach())
+        self.running_var.assign((1 - self.momentum) * self.running_var + self.momentum * prod(y.shape)/(prod(y.shape) - y.shape[1]) * batch_var.detach() )
```

Comment:
Hmm, are you sure this is right in general? You are subtracting just one of the channel axis from the product. Can you add a comment explaining this?

Reply:
Batchnorm2D is calculated per-dimension over the mini-batches, therefore you need to subtract when using Besel's correction not just 1 but the channel axis. In test_nn.py it is tested with a channel size of 4, but I tested it also with other values and it passed those as well.

Reply:
Comment + add those tests and I think it's good.